packages/datadog-plugin-bluebird/src/index.js

```diff
@@ -2,25 +2,56 @@
 
 const tx = require('../../dd-trace/src/plugins/util/promise')
 
-function createGetNewLibraryCopyWrap (tracer, config, shim) {
+const DD_LIB_COPIES = '_datadog_library_copies'
+
+function createGetNewLibraryCopyWrap (tracer, config, originalLib, shim) {
   return function wrapGetNewLibraryCopy (getNewLibraryCopy) {
     return function getNewLibraryCopyWithTrace () {
-      const result = getNewLibraryCopy.apply(this, arguments)
-      shim.wrap(result.prototype, '_then', tx.createWrapThen(tracer, config))
-      return result
+      const libraryCopy = getNewLibraryCopy.apply(this, arguments)
+      shim.wrap(libraryCopy.prototype, '_then', tx.createWrapThen(tracer, config))
+      addToLibraryCopies(originalLib.prototype, libraryCopy)
```

Comment:
Is there any particular reason to stash this on the prototype rather than the class/constructor? Just curious.

Reply:
nope, no good reason, updated to point to the constructor instead